### PR TITLE
EС-546

### PR DESCRIPTION
**Pull request by Bromel777**

### CODE REVIEW

File: src/main/scala/mvp2/actors/InfluxActor.scala [hunk 1]
```diff
@@ -4,50 +4,123 @@ import java.net.{InetAddress, InetSocketAddress}
 import akka.actor.Actor
 import com.typesafe.scalalogging.StrictLogging
 import mvp2.messages._
-import mvp2.utils.InfluxSettings
+import scala.concurrent.ExecutionContext.Implicits.global
+import mvp2.utils.{EncodingUtils, Settings}
 import org.influxdb.{InfluxDB, InfluxDBFactory}
 
-class InfluxActor(settings: InfluxSettings) extends Actor with StrictLogging {
+import scala.concurrent.duration._
```
Comment (GusevTimofey) [inline src/main/scala/mvp2/actors/InfluxActor.scala]: empty line

File: src/main/scala/mvp2/actors/Networker.scala [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
+
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.duration._
 import akka.actor.Props
 import mvp2.actors.Networker.Peer
+import mvp2.data.{Blockchain, KeyBlock}
 import mvp2.messages._
 import mvp2.utils.Settings
 
+import scala.collection.immutable.TreeMap
```
Comment (GusevTimofey) [inline src/main/scala/mvp2/actors/Networker.scala]: unused imports, empty lines

File: src/main/scala/mvp2/actors/Publisher.scala [hunk 3]
```diff
@@ -1,11 +1,16 @@
 package mvp2.actors
 
 import java.security.KeyPair
+
 import akka.util.ByteString
+
```
Comment (GusevTimofey) [inline src/main/scala/mvp2/actors/Publisher.scala]: unused imports, empty lines

File: src/main/scala/mvp2/actors/InfluxActor.scala [hunk 4]
```diff
@@ -4,50 +4,114 @@ import java.net.{InetAddress, InetSocketAddress}
...
+        case None =>
+          msgFromRemote = msgFromRemote - msgInfo._1 + msgInfo.copy(_2 = msgInfo._2 + (msg -> 1))
+          1
+      }
+      case None =>
+        msgFromRemote += (remote -> Map(msg -> 1))
+        1
+    }
+
+  def getToRemoteMsgIncrement(remote: InetSocketAddress, msg: String): Int =
```
Comment (ugulavaGeorge) [inline src/main/scala/mvp2/actors/InfluxActor.scala]: this and previous function are almost exactly the same. Are you sure, you can't refactor such stuff?

File: src/main/scala/mvp2/actors/Networker.scala [hunk 5]
```diff
@@ -27,21 +27,31 @@ class Networker(settings: Settings) extends CommonActor {
...
           (peers :+ msgFromRemote.remote).foreach(addPeer)
         case Ping =>
           logger.info(s"Get ping from: ${msgFromRemote.remote} send Pong")
-          context.actorSelection("/user/starter/networker/sender") ! SendToNetwork(Pong, msgFromRemote.remote)
+          context.actorSelection("/user/starter/blockchainer/networker/sender") !
+            SendToNetwork(Pong, msgFromRemote.remote)
         case Pong =>
           logger.info(s"Get pong from: ${msgFromRemote.remote} send Pong")
+        case Blocks(blocks) =>
+          logger.info(s"Receive blocks: ${blocks.mkString(",")} from remote: ${msgFromRemote.remote}")
```
Comment (ugulavaGeorge) [inline src/main/scala/mvp2/actors/Networker.scala]: there is nothing to do with arrived blocks?

File: src/main/scala/mvp2/actors/Receiver.scala [hunk 6]
```diff
@@ -1,29 +1,32 @@
 package mvp2.actors
 
 import java.net.{InetAddress, InetSocketAddress}
+
```
Comment (GusevTimofey) [inline src/main/scala/mvp2/actors/Receiver.scala]: empty line

**Review by GusevTimofey — APPROVED** [review]

**Review by ugulavaGeorge — APPROVED** [review]

**Review by kapinuss — APPROVED** [review]

**Pull request merged**